### PR TITLE
pybess: handle temporary HTTP2 disconnection

### PR DESCRIPTION
**Pull request by sangjinhan**
... during gRPC server launch. This race often happen `daemon start`; when bessd starts, if gRPC client connection is made prematurely, it is disconnected causing client to block forever.

### CODE REVIEW

File: pybess/bess.py [hunk 1]
```diff
@@ -192,20 +199,24 @@ def connect(self, grpc_url=DEF_GRPC_URL):
...
-                raise self.APIError(
-                    'Connection to {} failed'.format(grpc_url))
-            time.sleep(0.1)
+            self.status = None
+            self.peer = grpc_url
+            self.channel = grpc.insecure_channel(grpc_url)
+            self.channel.subscribe(self._update_status, try_to_connect=True)
+            self.stub = service_pb2.BESSControlStub(self.channel)
+
+            while not self.is_connected():
```
Comment (chris3torek) [inline pybess/bess.py]: The logic with these two nested `while not self.is_connected()` seems a bit tortured. Would it be possible to restructure this as:

    while not self.is_connected():
        if <some condition>:
            do the status, peer, channel, stub setup
        if <other conditions>: # or `elif`?
            disconnect, maybe raise error
        time.sleep(0.1)

?

Comment (shinae-woo) [reply pybess/bess.py]: Maybe changing the first `while not` can be replaced with `if` also good.

Comment (sangjinhan) [reply pybess/bess.py]: Thanks, revised. The `while` cannot be replaced with `if` in case of involuntary disconnection.

**Comment by shinae-woo** [comment]
image

**Pull request merged**